Source/EmbeddedBoundary/WarpXFaceExtensions.cpp

```diff
@@ -877,6 +992,75 @@ WarpX::ComputeEightWaysExtensions() {
                     borrowing_y_inds_pointer(i, j, k) = borrowing_y_inds + ps;
 
                     Sy_mod(i, j, k) = Sy(i, j, k);
+#ifdef WARPX_DIM_XZ
```

Comment:
For this large block: is there a way to reuse more code?
It looks to me like we might duplicate a lot of math here.

Reply:
I'm afraid this won't be possible. The problem is that in 3D the z-direction is stored in the third dimension of the arrays, while in XZ it is stored in the second dimension. 
For example, let's say we want to access the top-right neighbor's area. In the 3D code, it is going to be something like Sy(i+1, j, k+1), while in XZ it's going to be Sy(i+1, j+1, k). 

I couldn't see an easy way to work around this issue but I am open to ideas 😄

Reply:
Could you use constants for the index calculation/access and only put those in `#ifdef`s?

Reply:
You are making a really good point and, in general, I think I should try to reduce the code duplication in this file. We also discussed this with Remi in the past because in 3D we repeat the same operations 3 times and we just change the order of the indices.
The problem is mainly that lines like the following are difficult to generalize  

amrex::Real denom = local_avail(0, 1) * Sy(i - 1, j, k) +
                                     local_avail(2, 1) * Sy(i + 1, j, k) +
                                     local_avail(1, 0) * Sy(i, j - 1, k ) +
                                     local_avail(1, 2) * Sy(i, j + 1, k) +
                                     local_avail(0, 0) * Sy(i - 1, j - 1, k) +
                                     local_avail(2, 0) * Sy(i + 1, j - 1, k) +
                                     local_avail(0, 2) * Sy(i - 1, j + 1, k) +
                                     local_avail(2, 2) * Sy(i + 1, j + 1, k);

In principle, I could try to write a sort of decorator for the multifab class that depending on the dimension in consideration accesses the right indices. That might help to not expose this part of code to the choice of indices based on the dimension.

Reply:
Yes, we could generally write for instance a `constexpr` function that does the index permutation at compile time for the chosen dimensionality. That would come with zero runtime overhead.
Additional terms in sums like this could just be `#ifdef`d out.

Reply:
Sounds like a good strategy. I actually had already started that, but I missed that it should be `constexpr`
Thanks for the suggestion!